server/src/main/java/org/candlepin/controller/ManifestManager.java

```diff
@@ -0,0 +1,387 @@
+        ManifestFile manifest = manifestFileService.get(exportId);
+        if (manifest == null) {
+            throw new NotFoundException(
+                i18n.tr("Unable to find specified manifest by id: {0}", exportId));
+        }
+
+        if (!exportedConsumer.getUuid().equals(manifest.getTargetId())) {
+            throw new BadRequestException(
+                i18n.tr("Could not validate export against specifed consumer: {0}",
+                    exportedConsumer.getUuid()));
+        }
+
+        BufferedOutputStream output = null;
+        InputStream input = null;
+        try {
+            response.setContentType("application/zip");
+            response.setHeader("Content-Disposition", "attachment; filename=" + manifest.getName());
+
+            // NOTE: Input and output streams are expected to be closed by their creators.
+            input = manifest.getInputStream();
+            output = new BufferedOutputStream(response.getOutputStream());
+            int data = input.read();
+            while (data != -1) {
+                output.write(data);
+                data = input.read();
+            }
+            output.flush();
+        }
+        catch (Exception e) {
+            // Reset the response data so that a json response can be returned,
+            // by RestEasy.
+            response.setContentType("text/json");
+            response.setHeader("Content-Disposition", "");
+            throw new IseException(i18n.tr("Unable to download manifest: {0}", exportId), e);
+        }
+    }
+
+    /**
+     * Generates a manifest for the specifed consumer and stores the resulting file via the
+     * {@link ManifestFileService}.
+     *
+     * @param consumer the target consumer.
+     * @param cdnKey
+     * @param webAppPrefix
+     * @param apiUrl
+     * @return an {@link ExportResult} containing the details of the stored file.
+     * @throws ExportCreationException if there are any issues generating the manifest.
+     */
+    public ExportResult generateAndStoreManifest(Consumer consumer, String cdnKey, String webAppPrefix,
+        String apiUrl) throws ExportCreationException {
+        File export = null;
+        try {
+            poolManager.regenerateDirtyEntitlements(entitlementCurator.listByConsumer(consumer));
+            export = exporter.getFullExport(consumer, cdnKey, webAppPrefix, apiUrl);
+            ManifestFile manifestFile = storeExport(export, consumer);
+            sink.queueEvent(eventFactory.exportCreated(consumer));
+            return new ExportResult(consumer.getUuid(), manifestFile.getId());
+        }
+        catch (ManifestFileServiceException e) {
+            throw new ExportCreationException("Unable to create export archive", e);
+        }
+        finally {
+            // We no longer need the export work directory since the archive has been saved in the DB.
+            if (export != null) {
+                File workDir = export.getParentFile();
+                try {
```

Comment:
would it help for debugging if we allowed explicitly not cleaning up the directory upon failure?